librz/core/cmd/cmd_resize.c

```diff
@@ -4,21 +4,55 @@
 #include <string.h>
 #include <rz_core.h>
 
-static int rebase_helper(RzCore *core, ut64 oldbase, ut64 newbase) {
-	rz_debug_bp_rebase(core->dbg, oldbase, newbase);
-	rz_bin_set_baddr(core->bin, newbase);
-	rz_flag_move(core->flags, oldbase, newbase);
-	rz_core_bin_apply_all_info(core, rz_bin_cur(core->bin));
-	// TODO: rz_analysis_move :??
-	// TODO: differentiate analysis by map ranges (associated with files or memory maps)
-	return 0;
-}
+/**
+ * \brief rb - rebase all flags, binary, information, breakpoints, and analysis.
+ * \param core Rizin core.
+ * \param argc Count of command arguments.
+ * \param argv Vector containing command arguments.
+ * \return RZ_CMD_STATUS_OK or RZ_CMD_STATUS_ERROR.
+ */
+RZ_IPI RzCmdStatus rz_rebase_handler(RZ_BORROW RzCore *core, int argc, RZ_BORROW const char **argv) {
+
+	// get current file and current object
+	RzBinFile *bf = rz_bin_cur(core->bin);
+	if (!(bf && rz_file_exists(bf->file))) {
+		RZ_LOG_ERROR("Cannot open current RzBinFile.\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	RzBinObject *obj = rz_bin_cur_object(core->bin);
+	if (!obj) {
+		RZ_LOG_ERROR("Cannot retrieve current object.\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	// The file's "default" loading address, that baddr_shift is computed relative to.
+	ut64 static_base = obj->plugin->baddr(bf);
 
-RZ_IPI RzCmdStatus rz_rebase_handler(RzCore *core, int argc, const char **argv) {
-	ut64 oldbase = rz_num_math(core->num, argv[1]);
-	// old base = addr
-	// new base = core->offset
-	return rebase_helper(core, oldbase, core->offset);
+	// Compute current baddr_shift of analysis.
+	// Requires user input of the base address.
+	ut64 old_base = rz_num_math(core->num, argv[1]);
+	ut64 static_old_delta = old_base - static_base;
+
+	// perform actual rebase
+	RzList *sections_backup = rz_core_create_sections_backup(core);
+	if (!sections_backup) {
+		RZ_LOG_ERROR("Cannot create sections backup.\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
```

Comment:
Can you move this into the `rz_core_rebase_everything` function? If you do so you can make `rz_core_create_sections_backup` an internal function in cfile.c instead of an export RZ_API.